addons/crm/models/crm_lead.py

```diff
@@ -1202,6 +1324,333 @@ def get_import_templates(self):
+                    var_won_count.update({line['value']: won_count})
+
+                var_lost_count = vars_lost_count[line['variable']]
+                if not var_lost_count:
+                    var_lost_count = {line['value']: lost_count}
+                    vars_lost_count[line['variable']] = var_lost_count
+                else:
+                    var_lost_count.update({line['value']: lost_count})
+
+                var_total_count = vars_total_count[line['variable']]
+                if not var_total_count:
+                    var_total_count = {'won': won_count, 'lost': lost_count}
+                    vars_total_count[line['variable']] = var_total_count
+                else:
+                    var_total_count.update(
+                        {'won': var_total_count['won'] + won_count, 'lost': var_total_count['lost'] + lost_count})
+
+            # compute proba for each lead
+            if won and lost:
+                total = won + lost
+                p_won = won / total
+                p_lost = lost / total
+                for lead_id in leads_values_dict.keys():
+                    # 2. Compute won and lost score using each variable's individual probability
+                    s_lead_won = p_won
+                    s_lead_lost = p_lost
+                    for variable, value in leads_values_dict[lead_id]:
+                        if vars_won_count[variable]:
+                            for key, val in vars_won_count[variable].items():
+                                if str(value) == key:
+                                    if variable == 'stage_id':
+                                        s_lead_won *= vars_won_count[variable][key] / won
+                                        s_lead_lost *= vars_lost_count[variable][key] / lost
+                                    else:
+                                        s_lead_won *= vars_won_count[variable][key] / vars_total_count[variable]['won']
+                                        s_lead_lost *= vars_lost_count[variable][key] / vars_total_count[variable]['lost']
+
+                    # 3. Compute Probability to win
+                    lead_probabilities[lead_id] = 100 * s_lead_won / (s_lead_won + s_lead_lost)
+        return lead_probabilities
+
+    def _update_frequency_table(self, values_to_create, stage_ids, stage_sequences, company, team_id=None):
+        """ Create / update the frequency table in a cross company way.
+        But as the PLS start date depends on the company, we still need to store the company id,
+        to compute the probability per company, to be limited only on the time range configured on the company."""
+        frequencies = {}
+        pls_start_date = company.predictive_lead_scoring_start_date.strftime('%Y-%m-%d')
+        team_condition = 'and l.team_id = %s' % int(team_id) if team_id else 'and l.team_id is null'
+
+        def _increment_frequency(field, value, won, lost):
+            if field not in frequencies:
```

Comment:
You should probably init the 'frequencies' dict with an empty dict for each field so you avoid that initial if statement.